client/src/api.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @author: Artur Komoter
+ */
+
+exports.submitDocument = (route, body, redirectHandler) => {
+    let url = 'http://localhost:3001' + route;
+    fetch(url, {
+        method: 'POST',
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body)
+    }).then(response => response.json())
+        .then(redirectHandler)
+        .catch(console.log);
+};
+
+exports.handleInputChange = function(event) {
```

Comment:
Unfortunately due to arrow function's nature, `this` cannot be bound to any other object. Same goes for other functions that use `this`.